core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -177,6 +177,11 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(index);
     }
 
+    @Override
+    public Task createTask(long id, String type, String action) {
+        return new ReplicationTask(id, type, action, this::getDescription, getParentTaskNode(), getParentTaskId());
```

Comment:
The task management is opt-out for now. We can change the default implementation to make it opt-in, but I think that to get the most benefit we should enable it for most if not all actions. We discussed the getDescription handling in other PRs and the main problem with the current default implementation is that while we indeed evaluate it lazily, we also hold the reference to the entire request for longer than otherwise needed, which can be also bad for large requests. So, I am considering making the default implementation to be just a string instead of a string provider (which should have been supplier and not provider anyway) with default value of `""` instead of `TransportRequest#toString()`